ypathgen/pathgen.go

```diff
@@ -291,18 +303,63 @@ func (cg *GenConfig) GeneratePathCode(yangFiles, includePaths []string) (*Genera
 		if cg.GenerateWildcardPaths {
 			listBuilderKeyThreshold = cg.ListBuilderKeyThreshold
 		}
-		structSnippet, es := generateDirectorySnippet(directory, directories, schemaStructPkgAccessor, cg.PathStructSuffix, listBuilderKeyThreshold, cg.GenerateWildcardPaths, cg.SimplifyWildcardPaths)
+		fakeRootPkgName := defaultFakeRootPkgName
+		if cg.FakeRootPackageName != "" {
+			fakeRootPkgName = cg.FakeRootPackageName
+		}
+
+		structSnippet, es := generateDirectorySnippet(directory, directories, schemaStructPkgAccessor, cg.PathStructSuffix, listBuilderKeyThreshold, cg.GenerateWildcardPaths, cg.SimplifyWildcardPaths, cg.SplitByModule, cg.TrimOCPackage, cg.PackageName, fakeRootPkgName)
 		if es != nil {
 			errs = util.AppendErrs(errs, es)
 		}
-		structSnippets = append(structSnippets, structSnippet)
+		structSnippets = append(structSnippets, structSnippet...)
+	}
+
+	// Group snippets by package and compute their deps.
+	packages := map[string]*GeneratedPathCode{}
+	for _, snippet := range structSnippets {
+		if _, ok := packages[snippet.Package]; !ok {
+			packages[snippet.Package] = &GeneratedPathCode{
+				Deps: make(map[string]bool),
+			}
+		}
+		packages[snippet.Package].Structs = append(packages[snippet.Package].Structs, snippet)
+		for _, d := range snippet.Deps {
+			packages[snippet.Package].Deps[d] = true
+		}
+	}
+	for name, p := range packages {
+		err := writeHeader(yangFiles, includePaths, name, cg, p)
+		util.AppendErr(errs, err)
 	}
 	genCode.Structs = structSnippets
 
 	if len(errs) == 0 {
 		errs = nil
 	}
-	return genCode, nodeDataMap, errs
+	return packages, nodeDataMap, errs
+}
+
+// packageNameReplacePattern matches all characters allowed in yang modules, but not go packages.
+var packageNameReplacePattern = regexp.MustCompile("[._-]")
+
+// goPackageName returns the go package to use when generating code for the input Directory.
+// If splitByModule is false, the pkgName is always returned. Otherwise,
+// a transformed version of the module that the directory belongs to is returned.
+func goPackageName(dir *ygen.Directory, splitByModule, trimOCPkg bool, pkgName, fakeRootPkgName string) string {
```

Comment:
Could you also mention the purpose of trimOCPkg and fakeRootPkgName?

Reply:
added